src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqOutputChannel.cs

```diff
@@ -0,0 +1,133 @@
+            RabbitMqConnectionSettings connectionSettings,
+            System.ServiceModel.Channels.MessageEncoder encoder)
+            : base(factory)
+        {
+            _parent = factory;
+            _rabbitMqClient = rabbitMqClient;
+            _connectionSettings = connectionSettings;
+            _encoder = encoder;
+            _via = _connectionSettings.BaseAddress;
+            _baseAddress = new System.ServiceModel.EndpointAddress(_via);
+        }
+
+        System.ServiceModel.EndpointAddress System.ServiceModel.Channels.IOutputChannel.RemoteAddress => _baseAddress;
+
+        Uri System.ServiceModel.Channels.IOutputChannel.Via => _via;
+
+        public override T GetProperty<T>()
+        {
+            if (typeof(T) == typeof(System.ServiceModel.Channels.IOutputChannel))
+            {
+                return (T)(object)this;
+            }
+
+            T messageEncoderProperty = _encoder.GetProperty<T>();
+            if (messageEncoderProperty != null)
+            {
+                return messageEncoderProperty;
+            }
+
+            return base.GetProperty<T>();
+        }
+
+        /// <summary>
+        /// Open the channel for use. We do not have any blocking work to perform so this is a no-op
+        /// </summary>
+        protected override void OnOpen(TimeSpan timeout)
+        { }
+
+        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            return Task.CompletedTask;
+        }
+
+        protected override void OnEndOpen(IAsyncResult result)
+        { }
+
+        protected override void OnAbort()
+        { }
+
+        protected override void OnClose(TimeSpan timeout)
+        {
+            _rabbitMqClient.Close();
+        }
+
+        protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
+        { return Task.CompletedTask; }
+
+        protected override void OnEndClose(IAsyncResult result)
+        { }
+
+        /// <summary>
+        /// Address the Message and serialize it into a byte array.
+        /// </summary>
+        private ArraySegment<byte> EncodeMessage(System.ServiceModel.Channels.Message message)
+        {
+            try
+            {
+                _baseAddress.ApplyTo(message);
+                return _encoder.WriteMessage(
+                    message,
+                    (int)_parent.Transport.MaxReceivedMessageSize,
+                    _parent.BufferManager);
+            }
+            finally
+            {
+                // We have consumed the message by serializing it, so clean up
+                message.Close();
+            }
+        }
+
+        public void Send(System.ServiceModel.Channels.Message message)
+        {
+            var messageBuffer = EncodeMessage(message);
+
+            try
+            {
+                _rabbitMqClient.BasicPublish(
+                    exchange: _connectionSettings.Exchange,
```

Comment:
Check IModel.IsOpen before trying to send. What conditions can the RabbitMqClient end up not open? If it's something that can only be manually done, then you should Abort() the channel and throw something that derives from CommunicationException (or that exception itself).